statefun-flink/statefun-flink-core/src/main/java/org/apache/flink/statefun/flink/core/functions/Reductions.java

```diff
@@ -145,4 +142,21 @@ void processEnvelopes() {
       // TODO: consider preemption if too many local messages.
     }
   }
+
+  private static boolean useMultiplexedState(KeyedStateBackend<?> keyedStateBackend) {
+    final String backendClassName = keyedStateBackend.getClass().getName();
+
+    // TODO this is fragile and error-prone to classname changes, but we're doing this
+    // TODO to avoid additional dependencies on the Flink state backends
+    // TODO ideally, we should revisit how configuration is being passed to the
+    // TODO operators to be available at runtime
+    if (backendClassName.equals("org.apache.flink.runtime.state.heap.HeapKeyedStateBackend")) {
+      return false;
+    }
+    if (backendClassName.equals(
+        "org.apache.flink.contrib.streaming.state.RocksDBKeyedStateBackend")) {
```

Comment:
@tzulitai I think that we only need this branch, 
and this method can be simplified to simply:
`return backendClassName.equals(
         "org.apache.flink.contrib.streaming.state.RocksDBKeyedStateBackend")`

I think that we shouldn't care about any other state backends.

In addition, can a state backend be NULL, or is there a default toy state backend that is used for the mini cluster or something a long these lines?

Reply:
Ok, I actually see that the build fails due to this :-)
I'd vote to only focus on RocksDB.

Reply:
@igalshilman, the statebackend cannot be null. The default is MemoryStateBackend which is the same implementation as FsStateBackend but checks are stored in JM memory.

Reply:
@igalshilman 👍 Makes total sense. Will simply this branch.

^ the state backend is never `null` up to this point.